mm2src/mm2_bitcoin/script/src/sign.rs

```diff
@@ -111,6 +112,8 @@ impl From<TransactionInput> for UnsignedTransactionInput {
     fn from(i: TransactionInput) -> Self {
         UnsignedTransactionInput {
             previous_output: i.previous_output,
+            // DISCUSS: Would setting dummy script like this have any implications?
+            prev_script: Vec::new().into(),
```

Comment:
I have doubts about adding prev_script into UnsignedTransactionInput. Actually UnsignedTransactionInput is a proto for creating signed tx inputs (and prev_script is not part of it). Also with that we have two signing paths:
one when we call sign_tx() with the new interface w/o prev_script,
second when we use calc_and_sign_sighash() and here we do not use prev_script from UnsignedTransactionInput but pass it as a param. 
This looks for me as kind of ambiguous use of prev_script. I think better to have same behaviour: either use UnsignedTransactionInput::prev_script or not.

Reply:
I think I understood you here. Do you mean we could omit the `prev_script` from `calc_and_sign_sighash` since we can deterministically find it using `signer` + `input_index` (= `UnsginedTransactionInput`)??

Reply:
>  Do you mean we could omit the prev_script from calc_and_sign_sighash since we can deterministically find it using signer + input_index (= UnsginedTransactionInput)??

yes I think we should use one way to get prev_script. If you put it in UnsignedTransactionInput let's use it from there. Having options when it could be either in UnsignedTransactionInput or obtained differently complicates signing code IMO

Reply:
ummmm, I found an occurrence here:

https://github.com/mariocynicys/adex/blob/f07e720ad8d63c189d31200054f8e382b862aa98/mm2src/coins/utxo_signer/src/with_key_pair.rs#L178-L179

where the script stored in `UnsginedTransactionInput` is only used for validation while another (different) script is used for producing the signature.

For this case I think it's valid to force supply the `script` in `calc_and_sign_sighash`. It's gonna be redundant in most cases but necessary for other.
what do you think @dimxy ?

Reply:
I agree, let's have calc_and_sign_sighash as a low-level function that does input signing job and has only needed params

Reply:
> I think we can get the `scriptcode` from the `prev_script`

Editing the `prev_script` converting it from `p2wpkh` to `p2pkh`? I would be afraid playing with opcodes, huh

I think I spotted also other occurrences where our `prev_script` is p2sh script hash and we need to sign using the `redeem_script`, this one has no way back from `prev_script` to `scriptcode`.

Reply:
Are we closing this conversation for now? No changes to do?

(btw I guess we have more things to refactor in utxo signing, like:
We have inputs_infos and outputs_infos in UtxoSignTxParams but they are used in trezor signing only.
In p2pkh case we have duplication of prev_script in signer and output_script params of calc_and_sign_sighash(). Maybe we should look at all the structs and refactor them. 
Better to make a dedicated issue for that)

Reply:
Yeah I guess so, since `calc_and_sign_sighash` need to stay as it is because of the since the script to sign isn't always the `output_script` we found on the previous transaction (e.g. p2wpkh & p2sh).

Yeah we might want to refactor those to make use of the prev_script/output_script found on the signer.